src/extension.ts

```diff
@@ -292,7 +298,15 @@ export async function activate(extensionContext: extensionApi.ExtensionContext):
           }
         }
       }
-    );
+    ).then(() => false
+    ).catch(() => true); // required to force Promise.all() call keep waiting for all not failed calls
+  
+    const failed = await Promise.all([registryAccess,vmActivation]); // wait for all fail or pass
+    
+    if (failed.find(fail => fail)) {
+      removeSession(currentSession.id); // if at least one fail, remove session
+    }
```

Comment:
the code is OK but I think strange to have 'true' being the case where it failed and false being the case where it succeed.

Usually true = success, false = error

Reply:
That is because I am looking for failure not for success.

Reply:
You could look for failure using false

Reply:
I mean array of success instead of array of failures